bazel/repository_locations.bzl

```diff
@@ -304,30 +368,47 @@ REPOSITORY_LOCATIONS = dict(
         # Only allow peeking at fuzzer related files for now.
         strip_prefix = "compiler-rt-9.0.0.src/lib",
         urls = ["http://releases.llvm.org/9.0.0/compiler-rt-9.0.0.src.tar.xz"],
+        use_category = ["build"],
     ),
     fuzzit_linux = dict(
         sha256 = "9ca76ac1c22d9360936006efddf992977ebf8e4788ded8e5f9d511285c9ac774",
         urls = ["https://github.com/fuzzitdev/fuzzit/releases/download/v2.4.76/fuzzit_Linux_x86_64.zip"],
+        use_category = ["build", "test"],
     ),
     upb = dict(
         sha256 = "e9f281c56ab1eb1f97a80ca8a83bb7ef73d230eabb8591f83876f4e7b85d9b47",
         strip_prefix = "upb-8a3ae1ef3e3e3f26b45dec735c5776737fc7247f",
         # 2019-11-19
         urls = ["https://github.com/protocolbuffers/upb/archive/8a3ae1ef3e3e3f26b45dec735c5776737fc7247f.tar.gz"],
+        use_category = ["api"],
     ),
     kafka_source = dict(
         sha256 = "e7b748a62e432b5770db6dbb3b034c68c0ea212812cb51603ee7f3a8a35f06be",
         strip_prefix = "kafka-2.4.0/clients/src/main/resources/common/message",
         urls = ["https://github.com/apache/kafka/archive/2.4.0.zip"],
+        use_category = ["dataplane"],
     ),
     kafka_server_binary = dict(
         sha256 = "b9582bab0c3e8d131953b1afa72d6885ca1caae0061c2623071e7f396f2ccfee",
         strip_prefix = "kafka_2.12-2.4.0",
         urls = ["http://us.mirrors.quenda.co/apache/kafka/2.4.0/kafka_2.12-2.4.0.tgz"],
+        use_category = ["test"],
     ),
     kafka_python_client = dict(
         sha256 = "454bf3aafef9348017192417b7f0828a347ec2eaf3efba59336f3a3b68f10094",
         strip_prefix = "kafka-python-2.0.0",
         urls = ["https://github.com/dpkp/kafka-python/archive/2.0.0.tar.gz"],
+        use_category = ["test"],
     ),
 )
+
+def _strip_annotations():
+    stripped = dict(REPOSITORY_LOCATIONS_ANNOTATED)
+    for key, value in stripped.items():
+        value.pop("use_category", None)
+
+    return stripped
+
+# REPOSITORY_LOCATIONS is used to initialize http_archive which produces errors for annotation
```

Comment:
Can we filter at point-of-use rather than here? Ideally this file has minimal logic.

Reply:
done